### PR TITLE
Fix msgpack encoding/raw handling for older versions

### LINKED ISSUE #57501: [BUG] msgpack.py: 128: DeprecationWarning: encoding is deprecated, Use raw=False instead
**Description**
A lot of my minions are logging the message

```
[WARNING ] /usr/lib/python3.6/site-packages/salt/utils/msgpack.py:128: DeprecationWarning: encoding is deprecated, Use raw=False instead.
```

This issue looks related to Issue #51398, but is surfacing at another location

**Setup**
N/A

**Steps to Reproduce the behavior**
Control reaches https://github.com/saltstack/salt/blob/master/salt/serializers/msgpack.py#L49 with salt.utils.msgpack.version >= (0.5.2)

In this case, options.setdefault() will set encoding='utf-8', but encoding is deprecated asmsgpack version 0.5.2. The options.setdefault() call should be setting raw=False in this case.

**Expected behavior**
salt/serializers/msg.py.py doesn't log DeprecationWarnings

**Screenshots**
N/A

**Versions Report**
<details><summary>salt --versions-report</summary>
(Provided by running salt --versions-report. Please also mention any differences in master/minion versions.) 

```
Salt Version:
           Salt: 3000.3

Dependency Versions:
           cffi: Not Installed
       cherrypy: unknown
       dateutil: 2.4.2
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
         Jinja2: 2.8.1
        libgit2: Not Installed
       M2Crypto: 0.35.2
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.6.2
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 3.6.8 (default, Mar 30 2020, 17:04:00)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 15.3.0
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.1.4

System Versions:
           dist: centos 7.8.2003 Core
         locale: UTF-8
        machine: x86_64
        release: 4.4.131-1.el7.centos.x86_64
         system: Linux
        version: CentOS Linux 7.8.2003 Core

```
</details>

**Additional context**

It looks like the fix should look something like the block at https://github.com/saltstack/salt/blob/master/salt/transport/ipc.py#L179

### PR DESCRIPTION
**Pull request by oeuftete**
### What does this PR do?

This is an update of @jude's work in #57502, meant to address the outstanding change requests.

### What issues does this PR fix or reference?
Fixes: #57501

### Previous Behavior

A deprecation warning was logged in the minion logs.

### New Behavior

No deprecation warning is logged.

### Merge requirements satisfied?

- n/a Docs
- [x] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?

No

### CODE REVIEW

File: salt/utils/msgpack.py [hunk 1]
```diff
@@ -82,8 +80,10 @@ def _sanitize_msgpack_unpack_kwargs(kwargs):
     https://github.com/msgpack/msgpack-python/blob/master/ChangeLog.rst
     """
     assert isinstance(kwargs, dict)
+    if version >= (0, 5, 2):
+        kwargs.setdefault("raw", False)
+        kwargs.setdefault("encoding", "utf-8")
     if version >= (1, 0, 0):
-        kwargs.setdefault("raw", True)
```
Comment (Ch3LL) [inline salt/utils/msgpack.py:86]: why removing this? This is needed for msgpack versions 1.0.0

File: salt/serializers/msgpack.py [hunk 2]
```diff
@@ -46,7 +39,6 @@ def _serialize(obj, **options):
     def _deserialize(stream_or_string, **options):
         try:
             options.setdefault("use_list", True)
-            options.setdefault("raw", False)
```
Comment (Ch3LL) [inline salt/serializers/msgpack.py:49]: why are we removing the options here and on https://github.com/saltstack/salt/pull/58658/files#diff-0d5904efab02aab32413f541cbd67382L66 ? Different versions of msgpack will support different args.

File: salt/utils/msgpack.py [hunk 3]
```diff
@@ -82,8 +80,10 @@ def _sanitize_msgpack_unpack_kwargs(kwargs):
     https://github.com/msgpack/msgpack-python/blob/master/ChangeLog.rst
     """
     assert isinstance(kwargs, dict)
+    if version >= (0, 5, 2):
```
Comment (Ch3LL) [inline salt/utils/msgpack.py:83]: Would be best to define this as versions >= 0.5.2 and <1.0.0 to ensure we do not hit this if statement when running versions >= 1.0.0

**Review by Ch3LL — CHANGES_REQUESTED** [review]

**Comment by oeuftete** [comment]
The work I added was overreaching and wrong.  (Other than that...)

#57501 appears to have been fixed as a result of other improvements, so closing this one up.

**Pull request closed without merging**